internal/transformer-client/client.go

```diff
@@ -27,73 +49,211 @@ type ClientConfig struct {
 	PickerType    string        // power_of_two(default), round_robin, least_loaded_random, least_loaded_round_robin, random
 	Recycle       bool          // false
 	RecycleTTL    time.Duration // 60s
+
+	// Configuration for retryable HTTP client in case of [X-Rudder-Should-Retry: true] HTTP 503 responses
+	RetryRudderErrors struct {
+		Enabled         bool          // true
+		MaxRetry        int           // -1 - no limit
+		InitialInterval time.Duration // 1s
+		MaxInterval     time.Duration // 30s
+		MaxElapsedTime  time.Duration // 0s - no limit
+		Multiplier      float64       // 2.0
+	}
 }
 
 type Client interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
 func NewClient(config *ClientConfig) Client {
-	transport := &http.Transport{
-		DisableKeepAlives:   true,
-		MaxConnsPerHost:     100,
-		MaxIdleConnsPerHost: 10,
-		IdleConnTimeout:     30 * time.Second,
+	if config == nil {
+		return newRetryableHTTPClient(buildDefaultClient(), nil)
```

Comment:
why do we need to handle this scenario (nil ClientConfig)? We are not actually using it